corehq/apps/hqadmin/views/data.py

```diff
@@ -105,3 +112,22 @@ def check_form_for_errors(form, form_doc):
                 raw_data = repr(raw_data)
 
     return errors, raw_data
+
+
+@require_superuser
+def download_blob(request):
+    """Pairs with the get_download_url utility and command"""
+    key = request.GET.get("key")
+    try:
+        meta = BlobMeta.objects.partitioned_get(partition_value=key, key=key)
+    except BlobMeta.DoesNotExist as e:
+        raise Http404()
+    blob = get_blob_db().get(meta=meta)
```

Comment:
I initially made this view accept `key` and `type_code`, which can be used to get blobs (and would make this view possibly useful for other use-cases), but I needed access to the `meta` object for `content_length`, `content_type`, and `name`.  I couldn't figure out a better way to access the meta than that `partitioned_get` - was kinda surprised it wasn't easily accessible by the `key` and `type_code`.

Reply:
`partition_value` (which is the value used for `parent_id`) is required because blob metadata is stored in the shard databases. It would not be possible to lookup by `key` and `type_code` without hitting every shard.